### ISSUE TITLE
viafree.se - '--all-last' strange download order

### ISSUE THREAD

**Issue by Sopor-** [spaam/svtplay-dl#737]
If i run
```
svtplay-dl -A -S -P hls --all-last=2 https://www.viafree.se/program/livsstil/lyxfallan/sasong-23
```
it will try to download episode 08 and 09 even if episode 10 is the last one.

If i instead run (without the --all-last=2)
```
svtplay-dl -A -S -P hls https://www.viafree.se/program/livsstil/lyxfallan/sasong-23
```
it will first try to download episode 01 and then episode 10 and then the rest of the episodes.
I use --all-last to avoid every error on existing files when i use -A.

When i use `--all-last=2` it should download episode 10 and then 09.

svtplay-dl 1.9.6-28-g403f375

**Comment by granbom** [comment]
I've noticed the same thing on various sites, that downloading all episodes is done in a weird order.
My latest example:
<pre>
svtplay-dl -S -A https://www.svtplay.se/video/9947081/hipp-hipp/hipp-hipp-sasong-3-avsnitt-6
</pre>

Started with S02E04 then S01E04, then S02E01 and so on.

I my opinion I think svtplay-dl should sort the list of episodes, and that there should be a switch where you can choose to download from S01E01 or the end.
Sometimes when downloading a whole series, you'd like it to start from the begininng (and then you can actually start watching before everything is downloaded) and sometimes you'd like the last episodes in a series, because you've seen the others already.

svtplay-dl 1.9.6-28-g403f375

**Comment by spaam** [comment]
the issue with sorting is that it thinks 10 is numbers so it comes after 1, like this:  ```[u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-1', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-10', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-2', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-3', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-4', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-5', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-6', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-7', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-8', u'https://www.viafree.se/program/livsstil/lyxfallan/sasong-23/avsnitt-9']```

**Comment by nonadz** [comment]
Im also having some issues with the --all-last=NN 

If i want the latest episode from example svtplay, i run 

`svtplay-dl --all-last=01 https://www.svtplay.se/armar-och-tarmar`

But then i get an error saying 
> ERROR: This mode is not supported anymore. need the url with the video

So i tried to run 

`svtplay-dl --all-last=01  https://www.svtplay.se/video/15049608/armar-och-tarmar/armar-och-tarmar-sasong-1-avsnitt-1?start=auto`

But then it only downloads the current episode (episode 1)

Are im doing something wrong here?

**Comment by Sopor-** [comment]
@nonadz I believe you need to add `-A` too. Try this:
`svtplay-dl -A --all-last=1 https://www.svtplay.se/armar-och-tarmar`

**Comment by Sopor-** [comment]
> the issue with sorting is that it thinks 10 is numbers so it comes after 1, like this:

@spaam `what thinks`, the site or svtplay-dl?

**Comment by granbom** [comment]
Googled "natural sort python" and found https://github.com/SethMMorton/natsort. Guess it will do the trick, unfortunally only for py3 and I don't have time to check it out now. But I guess there'll be more natural sort implementations for python out there.

**Comment by spaam** [comment]
@Sopor- , python  is sorting the string by text. that is why 10 comes after 1 and not 2.

**Comment by spaam** [comment]
this should be fixed in master b47ce3ba2664c61466a2e1339ea877e36eefe215
